### PR TITLE
use fetch API and return promises

### PR DESCRIPTION
**Pull request by nylki**
Hey there,
Great library! The direct mapping makes it very robust. However I find myself increasingly working with the relatively new [Promise API](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Promise). Thats why I transformed the relevant parts of the code in jshue to use the newer `fetch` API instead of `XMLHttpRequest`, which enables the return of promises. (See new Readme for example usage).

The fail/success callbacks have been removed, because errors can now simply be `catch`ed at the end of the promise.

I can understand if you prefer not not merge it, because of incompatibilities with older browsers or because you prefer the callback syntax.
Basically, I still wanted to give you a heads up that I did this fork and give you the option to merge this change if you like. Otherwise I'll publish the fork to npm under the name `jshue` for easy installation, unless you'd prefer that I use a different name for npm like `jshue-next` for better distinction. Just let me know.

Credit to your original project and the MIT licence remains of course.

### CODE REVIEW

**Comment by blargoner** [comment]
@nylki Thanks for this! I plan to review it this weekend.

File: src/jshue.js [hunk 1]
```diff
@@ -2,38 +2,23 @@
  * jsHue
  * JavaScript library for Philips Hue.
  *
- * @module jshue
- * @version 0.3.0
+ * @module jshue-next
```
Comment (blargoner) [inline src/jshue.js]: Please change to `jshue`

**Comment by nylki** [comment]
>  Thanks for submitting these changes!

And thank you for reviewing them so quickly :)

> When I originally created jsHue, I wanted the simplest possible set of dependencies so that it would run almost anywhere, which is why I went with the callback interface. However, that was a long time ago, and I think this is a natural and reasonable evolution of the library at this time.

I completely understand, I would most likely have decided in the same way 2-4 years ago considering the coverage of native Promise api and fetch at that time.

> I requested some minor changes to your code. Please let me know if you can make these changes, after which point I can merge.

I am happy to make the changes in the coming days, I'll keep you informed.

File: README.md [hunk 2]
```diff
@@ -35,37 +36,30 @@ var hue = jsHue();
...
+}
+).catch((err) => {console.log('Error finding bridges', err)});
 ```
 
-jsHue performs requests asynchronously and provides a callback interface. The
-failure callback is called if the XHR request fails, or if there is an error with
-the JSON serialization or deserialization. All API success and error results are
-returned in the success callback data. (This is because some API calls may return
-aggregated API success and error results.)
+jsHue-promise performs requests asynchronously and provides a Promise interface. You can `catch` errors as usual, like in the example below.
```
Comment (nylki) [inline README.md]: Sorry for my absence. We could potentially change the behaviour to check for `failure` hue API responses (in `_requestJson`) and throw an error in the promise, so that all errors are catchable and only `success` objects get returned in a `then`. Would that be preferable or better leave it as it is?

Comment (blargoner) [inline README.md]: @nylki No worries. I would prefer not to do that at this time, because it is possible for the bridge to respond with a mix of successes and failures, like:
```json
[
	{
		"error": {
			"type": 201,
			"address": "/lights/1/state/bri",
			"description": "parameter, bri, is not modifiable. Device is set to off."
		}
	},
	{
		"success": {
			"/lights/1/state/on": false
		}
	}
]
```

and I don't want jsHue to express an opinion on whether that counts as success or failure at the library level, as it's really for the user to say. I'd prefer to keep jsHue's responsibility limited to serialization/deserialization and request/response processing, and pass all results back to the user.

Comment (nylki) [inline README.md]: I wasn't aware of the possibility of mixed responses. Then it makes totally sense to keep it as it is. :)

**Comment by nylki** [comment]
I pushed the changed your asked for. Let me know if I missed something; I also added the "Contributors" section in the Readme.md but left it empty for you to fill in your preferred style.

**Comment by blargoner** [comment]
@nylki Thanks! I will take a look in the next few days.

**Comment by blargoner** [comment]
@nylki Sorry for the delay. Reviewed, made some additional changes, merged, and deployed in a new release. Thanks again.

**Comment by nylki** [comment]
Was a pleasure!
